File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys

N = int(sys.stdin.readline())

A = list(map(int, sys.stdin.readline().split()))

M = int(sys.stdin.readline())

B = list(map(int, sys.stdin.readline().split()))


# 이진 탐색 알고리즘 사용하기

A.sort()

for i in range(M):
    start = 0
    end = len(A) - 1

    x = 0

    while(start <= end):
        mid = (start + end) // 2
        if(B[i] == A[mid]):
            x = 1
            break
        elif(B[i] < A[mid]):
            end = mid - 1
            
        else:
            start = mid + 1
    print(x)

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.*;

public class Main {

    static int N;
    static int M;
    static int A[];
    static int B[];
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        N = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(st.nextToken());
        }

        M = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        B = new int[M];
        for (int i = 0; i < M; i++) {
            B[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(A);

        for (int i = 0; i < M; i++) {
            System.out.println(comp(B[i]));
        }
    }

    public static int comp(int a){
        int start = 0;
        int end = A.length - 1;
        boolean x = false;
        while(start <= end){
            int mid = (end + start) / 2;

            if(A[mid] > a){
                end = mid - 1;
            }else if(A[mid] < a){
                start = mid + 1;
            }else{
                x = true;
                break;
            }
        }
        
        if(x){
            return 1;
        }else{
            return 0;
        }
    }
}
